src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -172,17 +205,37 @@ function ScatterplotViz(props: Props) {
     [updateVizConfig]
   );
 
+  // set valueSpec as Raw when yAxisVariable = date
+  useEffect(() => {
```

Comment:
(To be addressed after Variable Coverage is merged.)

The linter is warning about missing dependencies for this effect:

```
Line 215:6:  React Hook useEffect has missing dependencies: 'findVariable' and 'updateVizConfig'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

Simply adding `findVariable` and `updateVizConfig` to the dependency list isn't ideal, as that could cause `updateVizConfig({ valueSpecConfig: 'Raw' })` to be run unnecessarily.

I propose deleting this effect and moving the "if the yAxisVariable is of type 'date', call `updateVizConfig({ valueSpecConfig: 'Raw' })`" logic to the definition of the `handleInputVariableChange` callback.

Once Variable Coverage is merged, it will be clearer how the code will need to change.

Reply:
@jtlong3rd thank you for review and suggestion 👍 I missed that linter message.
I tested your approach like this at `handleInputVariableChange` (will change `findVariable` to `findEntityAndVariable` after your merge):
```
      updateVizConfig({
        xAxisVariable,
        yAxisVariable,
        overlayVariable,
        facetVariable,
        valueSpecConfig: (findVariable(yAxisVariable)?.type === 'date')
          ? 'Raw'
          : vizConfig.valueSpecConfig
      });
```

Basically it works as well, however, I still get a similar error, `React Hook useCallback has missing dependencies: 'findVariable' and 'vizConfig.valueSpecConfig'...`

Reply:
Yes, you're right! It's OK add those missing dependencies to the `useCallback`. Because this is a `useCallback` and not a `useEffect`, `updateVizConfig` won't actually get called until `handleInputVariableChange` gets invoked.

Reply:
@jtlong3rd Aha I misunderstood your intention 😄 Yep I will do that after your merge. Thanks again 👍